Feature/detector gpt

chrstrom opened the pull request:
Addresses #2 and #3 

Default parameters still need to be set, and tuning is TBD. @Bruglen add a few more reviewers pls 🐱

chrstrom committed:
Change tuning based on office environment

Signed-off-by: Christopher Strøm <detoxte@gmail.com>

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Hotfix: Changes that were overwritten at the Stormtrooper incident

Signed-off-by: Christopher Strøm <chstr@stud.ntnu.no>

chrstrom committed:
Merge branch 'feature/detector-gpt' of https://github.com/vortexntnu/vortex-sensor-fusion into feature/detector-gpt

chrstrom committed:
Dont use untested detectors

Signed-off-by: Christopher Strøm <chstr@stud.ntnu.no>

chrstrom committed:
Publish centroids in expected PDAF format

Signed-off-by: Christopher Strøm <chstr@stud.ntnu.no>

chrstrom committed:
Hotfixes for build

Signed-off-by: Christopher Strøm <chstr@stud.ntnu.no>

chrstrom committed:
Add updated db

Signed-off-by: Christopher Strøm <chstr@stud.ntnu.no>

chrstrom committed:
Hotfixes from stormtrooper testing

Signed-off-by: Christopher Strøm <chstr@stud.ntnu.no>

chrstrom committed:
Start dynamic reconfigure implementation

Signed-off-by: Christopher Strøm <detoxte@gmail.com>

chrstrom committed:
Setup configs and build for dynamic reconfigure

Signed-off-by: Christopher Strøm <detoxte@gmail.com>

chrstrom committed:
Finish dynamic reconfigure

Signed-off-by: Christopher Strøm <detoxte@gmail.com>

chrstrom committed:
Hotfix: Formatting

Signed-off-by: Christopher Strøm <detoxte@gmail.com>

chrstrom commented:
@Bruglen Could you put the steps to recreate the errors you got on Stormtrooper here?

Bruglen commented:
1) I turn on the lidar by plugging in the power
2) I build the lidar_ws
3) I source the workspace 
4) In the workspace i write: roslaunch pcl_detector detector.launch 


vortex:~/lidar_ws$ roslaunch pcl_detector detector.launch 

PARAMETERS
 * /os_cloud_node/tf_prefix: 
 * /os_node/imu_port: 0
 * /os_node/lidar_mode: 2048x10
 * /os_node/lidar_port: 0
 * /os_node/metadata: metadata.json
 * /os_node/replay: False
 * /os_node/sensor_hostname: os1-122007000043....
 * /os_node/timestamp_mode: 
 * /os_node/udp_dest: 
 * /pcl_detector/dbscan/epsilon: 1
 * /pcl_detector/dbscan/min_points: 100
 * /pcl_detector/detector_type: dbscan
 * /pcl_detector/euclidean/cluster_tolerance: 1
 * /pcl_detector/euclidean/min_points: 100
 * /pcl_detector/gmm/max_iterations: 100
 * /pcl_detector/gmm/num_clusters: 10
 * /pcl_detector/gmm/step_size: 0.01
 * /pcl_detector/leaf_size: 0.05
 * /pcl_detector/optics/epsilon: 1
 * /pcl_detector/optics/min_points: 100
 * /rosdistro: melodic
 * /rosversion: 1.14.13

NODES
  /
    os_cloud_node (ouster_ros/os_cloud_node)
    os_node (ouster_ros/os_node)
    pcl_detector_ros_node (pcl_detector/pcl_detector_ros_node)
    relay_imu (topic_tools/relay)
    relay_lidar (topic_tools/relay)

auto-starting new master
process[master]: started with pid [22170]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 84f221d8-c647-11ed-b151-f43909386a6d
process[rosout-1]: started with pid [22233]
started core service [/rosout]
process[os_node-2]: started with pid [22236]
process[os_cloud_node-3]: started with pid [22250]
process[relay_lidar-4]: started with pid [22252]
process[relay_imu-5]: started with pid [22267]
process[pcl_detector_ros_node-6]: started with pid [22268]
[ INFO ] [1679224599.984340905] [/os_node]: Client version: 0.3.0.dev2+a484e0c-release
[ INFO ] [1679224599.987255648] [/os_node]: Using automatic UDP destination
[ INFO ] [1679224599.987275719] [/os_node]: Waiting for sensor os1-122007000043.local to initialize ...
[ INFO ] [1679224600.023979932] [/pcl_detector_ros_node]: Configured dbscan with eps=1 and min_points=100

[ INFO ] [1679224620.497693898] [/os_node]: Sensor initialized successfully
[ INFO ] [1679224620.499677906] [/os_node]: Wrote metadata to metadata.json
WARNING: No data_format found.
WARNING: No lidar_origin_to_beam_origin_mm found.
[ INFO ] [1679224620.501483977] [/os_node]: Using lidar_mode: 2048x10
[ INFO ] [1679224620.501548168] [/os_node]: OS-1-16-A2 sn: 122007000043 firmware rev: v1.13.0
[pcl_detector_ros_node-6] process has died [pid 22268, exit code -11, cmd /home/vortex/lidar_ws/devel/lib/pcl_detector/pcl_detector_ros_node __name:=pcl_detector_ros_node __log:=/home/vortex/.ros/log/84f221d8-c647-11ed-b151-f43909386a6d/pcl_detector_ros_node-6.log].
log file: /home/vortex/.ros/log/84f221d8-c647-11ed-b151-f43909386a6d/pcl_detector_ros_node-6*.log

chrstrom committed:
Hotfix: Actually return the configured detector

Signed-off-by: Christopher Strøm <detoxte@gmail.com>

chrstrom committed:
Remove GMM and OPTICS

Signed-off-by: Christopher Strøm <detoxte@gmail.com>

chrstrom commented:
Fixed the error you found and also deleted GMM and OPTICS. They are in the commit history anyways so, and I don't think the implementation should be part of this merge. Maybe another time

ronjakr commented on detectors/pcl_detector/params/example_params.yaml:
```diff
@@ -0,0 +1,23 @@
+pcl_detector:
+  leaf_size: 0.05
+  detector_type: "dbscan" # one of "dbscan", "euclidean", "gmm", "optics"
```
maybe remove gmm and optics stuff since the gmm and optics files are deleted

ronjakr reviewed (approved):
seems to make sense

hannahcl committed:
tuning DBSCAN. See wiki for comments
